modules/service/publish-service.js

```diff
@@ -135,6 +146,10 @@ class PublishService {
 
         return status;
     }
+
+    async sleepForMilliseconds(milliseconds) {
```

Comment:
Can we move this to utilities, or to check how we performed sleep on other places in code